cmd/cronosd/cmd/fix-unlucky-tx.go

```diff
@@ -305,21 +349,27 @@ func newTxIndexer(config *tmcfg.Config, chainID string) (txindex.TxIndexer, erro
 	}
 }
 
-func findUnluckyTx(blockResult *tmstate.ABCIResponses) int {
+func (db *tmDB) findUnluckyTx(blockResult *tmstate.ABCIResponses, block *tmtypes.Block) (int, error) {
 	for txIndex, txResult := range blockResult.DeliverTxs {
 		if rpc.TxExceedsBlockGasLimit(txResult) {
-			return txIndex
+			tx := block.Txs[txIndex]
+			txHash := tx.Hash()
+			indexed, err := db.txIndexer.Get(txHash)
+			if err != nil {
+				return -1, err
+			}
+			if indexed != nil && txResult.Code == 0 && txResult.Code == indexed.Result.Code {
+				fmt.Printf("skip %x at %d", txHash, txIndex)
+				continue
```

Comment:
@yihuang will we have concurrent issue when 2 blocks check txIndexer together

Reply:
I think no